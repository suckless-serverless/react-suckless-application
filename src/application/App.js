import Vue from 'vue'
import App from './App.vue'

export default class VueApplication {
  start (root) {
    new Vue({
      render: h => h(App)
    }).$mount('#' + root)
  }
}
