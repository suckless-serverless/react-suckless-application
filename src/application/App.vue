<template>
  <div @click="click">
    <img alt="Vue logo" height="200px" src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Vue.js_Logo_2.svg/1200px-Vue.js_Logo_2.svg.png" />
    {{ name }} => {{ number}}
  </div>
</template>

<script>

export default {
  name: 'App',
  data () {
    return {
      name: 'pepe',
      number: 0
    }
  },
  methods: {
    click () {
      this.number++
    }
  }
}
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
