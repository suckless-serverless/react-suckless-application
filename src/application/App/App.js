import React from 'react'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.click = this.click.bind(this)
    this.state = {
      name: 'Number of clicks',
      number: 0
    }
  }
  click () {
    let { number } = this.state
    number++
    this.setState({ number })
  }
  render () {
    return <div>
      <img src="https://www.cybelesoft.com/wp-content/uploads/2017/08/reactlogo2-150x150.png" alt=""/>
      <button onClick={this.click}>Click Me</button>
      { this.state.name } =&gt;  { this.state.number }
      <div id="reactcontainer">
        you contained app goes here
      </div>
    </div>
  }
}

export default App
