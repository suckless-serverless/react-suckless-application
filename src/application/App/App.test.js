import React from 'react'
import renderer from 'react-test-renderer'
import App from './App.js'

describe('MessageList.test.js', () => {
  let app
  beforeEach(() => {
    app = renderer
      .create(<App />)
  })
  it('renders correctly', () => {
    const tree = app.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('clicks event', () => {
    let tree = app.toJSON()
    // manually trigger the callback
    tree.children[1].props.onClick()
    // re-rendering
    tree = app.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
