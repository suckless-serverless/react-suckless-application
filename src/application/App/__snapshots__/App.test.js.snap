// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.test.js clicks event 1`] = `
<div>
  <img
    alt=""
    src="https://www.cybelesoft.com/wp-content/uploads/2017/08/reactlogo2-150x150.png"
  />
  <button
    onClick={[Function]}
  >
    Click Me
  </button>
  Number of clicks
   =&gt;  
  1
  <div
    id="container"
  >
    you contained app goes here
  </div>
</div>
`;

exports[`MessageList.test.js renders correctly 1`] = `
<div>
  <img
    alt=""
    src="https://www.cybelesoft.com/wp-content/uploads/2017/08/reactlogo2-150x150.png"
  />
  <button
    onClick={[Function]}
  >
    Click Me
  </button>
  Number of clicks
   =&gt;  
  0
  <div
    id="container"
  >
    you contained app goes here
  </div>
</div>
`;
